acceptance/tests/resource/file/source_attribute.rb

```diff
@@ -182,43 +189,51 @@ class source_test_module {
   end
 end
 
-new_mod_manifest = <<EOF
-class source_test_module {
-#{mod_manifest_entry}
-}
-EOF
 apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
   file { '#{mod_manifest_file}':
     ensure => file,
     mode => '0644',
-    content => '#{new_mod_manifest}',
+    content => '#{mod_manifest}',
   }
 MANIFEST
 
 # TODO: Add tests for puppet:// URIs with multi-master/agent setups.
-step "when using a puppet://$server/ URI with a master/agent setup"
+step "When using puppet apply"
 agents.each do |agent|
   step "Setup testing local file sources"
   a_testdir = agent.tmpdir('local_source_file_test')
 
   source = "#{a_testdir}/source_mod/files/source"
-  target = "#{a_testdir}/target"
-
-  on agent, "mkdir -p #{File.dirname(source)}"
-  create_remote_file agent, source, 'Yay, this is the local file.'
+  checksums.each do |checksum_type|
+    target = "#{a_testdir}/target#{checksum_type}"
+    checksum = if checksum_type then "checksum => #{checksum_type}," else "" end
+
+    on agent, "mkdir -p #{File.dirname(source)}"
+    create_remote_file agent, source, 'Yay, this is the local file.'
+
+    step "Using a local file path (#{checksum})"
+    manifest = %(file { '#{target}': source => '#{source}', ensure => present, #{checksum} })
+    apply_manifest_on agent, manifest
+    on agent, "cat #{target}" do
+      assert_match(/Yay, this is the local file./, stdout, "FIRST: File contents not matched on #{agent}")
```

Comment:
The `lite` checksum types only compare the first 512 bytes of input, which is bigger than the file content we're managing. So in this test, `md5lite` is effectively the same as `md5`, etc. I think there is value in testing this combination, but I think we also want to test the case where the file content > 512. In that case, the `lite` checksums should only compare the first 512 bytes, ignoring whatever data comes after. You could even have the test modify `content[512]` and verify the `lite` checksums don't detect the change.